lib/segment/src/payload_storage/tests.rs

```diff
@@ -0,0 +1,106 @@
+use serde_json::json;
+
+use super::mmap_payload_storage::MmapPayloadStorage;
+use super::on_disk_payload_storage::OnDiskPayloadStorage;
+use super::simple_payload_storage::SimplePayloadStorage;
+use super::PayloadStorage;
+use crate::common::rocksdb_wrapper::open_db;
+use crate::types::Payload;
+
+fn test_trait_impl<S: PayloadStorage>(mut storage: S) {
```

Comment:
WDYT about adding `open` to the `PayloadStorage` interface?

This way we could test persistence here by dropping the storage and  then reopening it to check that the payloads are still present.